### DISCUSSION TITLE
Can't install KB5012170 in ed2k-ovmf in Qemu

### DISCUSSION THREAD

**Discussion by Elrondo46** [tianocore/edk2#3221] [General]
Hi,

I've Windows 11 virtualized and this update can't be install with this UEFI config in QEMU-KVM, think it's caused by actual UEFI firmware. In real machine no probs, in VMWare (UEFI) no probs.

There is a user have the same problem in Windows 10

https://www.linuxquestions.org/questions/showthread.php?p=6374147#post6374147

**Comment by ajfish** [comment]
If the issue is DBX related you may be able to manually provision your VM with the latest DBX [1]

[1] Info on getting the latest DBX file:
https://uefi.org/revocationlistfile

Thanks,

Andrew Fish

**Reply by Elrondo46** [reply]
How to do that ?

**Comment by ajfish** [comment]
Something like: https://projectacrn.github.io/latest/tutorials/waag-secure-boot.html

**Comment by MichaelTrip** [comment]
Will the DBX be updated in a later release of the ed2k-ovmf ?

**Comment by bentalexanderhaase** [comment]
Any updates on this issue? This seems to affect at least a few people and WIndows VMs with Secure Boot can't get past any update once `KB5012170` is in the update queue. Windows reboots, fails to apply this update and reverts. Any help would be appreciated, especially for the less UEFI literate sysadmins out there :blush:

**Reply by bentalexanderhaase** [reply]
Following: https://support.microsoft.com/en-us/topic/microsoft-guidance-for-applying-secure-boot-dbx-update-kb4575994-e3b9e4cb-a330-b3ba-a602-15083965d9ca which pretty much explained how you do #3276 reeks some overly informative error:

```ps
PS C:\Program Files\WindowsPowerShell\Scripts> Set-SecureBootUEFI -Name dbx -ContentFilePath .\content.bin -SignedFilePath .\signature.p7 -Time 2022-09-01T18:24:31Z -AppendWrite
Set-SecureBootUEFI : Unexpected Result, status error: 0xC000000D
At line:1 char:1
+ Set-SecureBootUEFI -Name dbx -ContentFilePath .\content.bin -SignedFi ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Microsoft.Secur...BootUefiCommand:SetSecureBootUefiCommand) [Set-SecureBootUEFI], StatusException
    + FullyQualifiedErrorId : SetFWVarFailed,Microsoft.SecureBoot.Commands.SetSecureBootUefiCommand
```

any help is greatly appreciated!

**Reply by svarshavchik** [reply]
> Following: https://support.microsoft.com/en-us/topic/microsoft-guidance-for-applying-secure-boot-dbx-update-kb4575994-e3b9e4cb-a330-b3ba-a602-15083965d9ca which pretty much explained how you do #3276 reeks some overly informative error:

Even that is way over my head. I'm confused about some fundamental principle here. qemu emulates VM hardware. OVMF is involved with this, in some way. The qemu VM's configuration includes a "OVMF_CODE.secboot.fd" file, on the host, amongst other things. That, pretty much, is the sum total of my knowledge of all things concerned with EFI and secureboot.

If the EFI voodoo that Microsoft has blacklisted is provided by the host's VM emulation, which involves OVMF, then how does doing anything within the guest, as described in that support article, update (in some form or fashion) this host-provided file?

And just to make things confusing: while I was trying to figuring out what's up with my Windows updates, one of the things that I came across is this https://www.bleepingcomputer.com/news/security/windows-kb5012170-secure-boot-dbx-update-may-fail-with-0x800f0922-error/ which mumbles something about insecure third part EFI images. No names mentioned there sound like "OVMF".

**Reply by bentalexanderhaase** [reply]
Thank you for you comment, at least I am not the only one struggling to understand what exactly is going on here. After roughly 10 hours of tinkering I found out that another version of tianocore/edk2/OVMF allows `KB5012170` to be successfully installed (manually via downloaded update file, as well as automatically via windows update service.

The [version which seems to work](http://download.proxmox.com/debian/pve/dists/bullseye/pve-no-subscription/binary-amd64/pve-edk2-firmware_3.20220526-1_all.deb) (just extract the necessary files from this `.deb` file) is from Proxmox VE 7.x. I simply copied over `OVMF_CODE_4M.secboot.fd` and `OVMF_VARS_4M.ms.fd` to my local Fedora 36 machine and booted my VM with the following XML configuration:

```xml
  <os>
    <type arch="x86_64" machine="pc-q35-6.2">hvm</type>
    <loader readonly="yes" secure="yes" type="pflash">/usr/share/edk2/ovmf/OVMF_CODE_4M.secboot.fd</loader>
    <nvram>/var/lib/libvirt/qemu/nvram/<vm_name>-OVMF_VARS_4M.ms.fd</nvram>
    [...]
  </os>
```

**:warning: This tripps Bitlocker (so make sure to have your recovery key ready if you use Bitlocker!).**

After booting, the update could be applied and the system can be rebooted which automatically re-enrolls the Bitlocker key onto the vTPM (if used).

Sadly this only fixes 1/3 updates for me. Now my system runs into a bootloop on two other updates [`KB5017308`](https://support.microsoft.com/en-us/topic/september-13-2022-kb5017308-os-builds-19042-2006-19043-2006-and-19044-2006-e4ea187e-28e8-4d4b-808b-2794babdce4c),[`KB5017500`](https://support.microsoft.com/en-us/topic/september-13-2022-kb5017500-cumulative-update-for-net-framework-3-5-4-8-and-4-8-1-for-windows-10-version-21h2-23323482-d4f8-4105-bdbb-4f7cf267fb66) -> `0x800f0845`. Of which `KB5017308` also seems to be somewhat security related.

I never had update issues with my Windows VMs but something seems to be off with the latest three Windows security updates from 08.2022 + 09.2022. 

---

*Keywords for future time travelers*

QEMU+KVM, [`KB5012170`](https://support.microsoft.com/en-us/topic/kb5012170-security-update-for-secure-boot-dbx-august-9-2022-72ff5eed-25b4-47c7-be28-c42bd211bb15) [`KB5017308`](https://support.microsoft.com/en-us/topic/september-13-2022-kb5017308-os-builds-19042-2006-19043-2006-and-19044-2006-e4ea187e-28e8-4d4b-808b-2794babdce4c),[`KB5017500`](https://support.microsoft.com/en-us/topic/september-13-2022-kb5017500-cumulative-update-for-net-framework-3-5-4-8-and-4-8-1-for-windows-10-version-21h2-23323482-d4f8-4105-bdbb-4f7cf267fb66), kvm Windows 10 VM, Windows 10 VM Bitlocker TPM update issues, `0x800f0845`

**Reply by hnz101** [reply]
> The [version which seems to work](http://download.proxmox.com/debian/pve/dists/bullseye/pve-no-subscription/binary-amd64/pve-edk2-firmware_3.20220526-1_all.deb) (just extract the necessary files from this `.deb` file) is from Proxmox VE 7.x. I simply copied over `OVMF_CODE_4M.secboot.fd` and `OVMF_VARS_4M.ms.fd` to my local Fedora 36 machine and booted my VM with the following XML configuration:

This fixed it for me, thank you! I am running an Windows 10 VM (no secure boot, no bitlocker!) on Bullseye and with this version KB5012170 finally installed without error.
I just switched out "OVMF_CODE.fd" with "OVMF_CODE_4M.fd" in my XML loader Parameter.

I already tried a newer Package from Debian testing/SID (ovmf_2022.08-1_all.deb), but that didn't worked out. I am also wondering why there isn't more information about this issue regarding running W10 on qemu/kvm.

**edit after @BentHaase reply**
It seems like I am actually using the debian OVMF. I didn't properly looked at the paths of the proxmox deb package files. **The problem just was not using the 4M Version of the Firmware.**

**Reply by bentalexanderhaase** [reply]
Likely because not too many people have noticed this issue so far. This also seems to only affect 'older' / longer running Windows 10 VMs that have been created before OVMF with 4M space was a thing (seems to be the default in Proxmox now). This issue just popped up on the 15th of September 2022 so maybe more people will notice these issues and land here. Maybe some don't care about failed updates in Windows VMs as long there is no policy enforcing it to have all updates by date x.

*Interestingly OVMF on Fedora 36 does not come with a 4M version (at least not obviously named like so), which is why I had to copy mine over from a Proxmox host. :thinking:*

Nonetheless with my particular system I have had a ton of issues with `KB5017308` now which seems to be impossible to install and after roughly 12h+ of debugging and reading Windows logs I have accepted my defeat and will surrender to re-install this machine as I can't reproduce this [issue with `KB5017308`] on a fresh install + upgrade from an older ISO.

Glad my findings helped you!

**Reply by svarshavchik** [reply]
I'm unfamiliar with these packaging details, and how exactly OVMF*fd files get built. I'm happy to create Fedora bug against their ed2k package, if I can have a cogent explanation of what needs to be done to build these images, that I can put into the bug.

**Reply by Markus-Go** [reply]
I had the same promblem installing KB5012170 on a KVM virtualized machine (Win10 Pro, no secure boot, no vTPM) on Debian 10.
Debian 10 does not ship with 4M OVMF files, only with the old 2M ones.

What worked for me:
* Download OVMF from Debian 11 (https://debian.pkgs.org/11/debian-main-arm64/ovmf_2020.11-2+deb11u1_all.deb.html)
* Extract OVMF_CODE_4M.fd and copy it to /usr/share/OVMF/
* Change KVM config to use it (I did not extract or change the NVRAM file, just used the old one)

**Reply by Elrondo46** [reply]
Changed to 4M files and update success.

**Reply by MichaelTrip** [reply]
Also worked for me. I grabbed the ovmf files from proxmox and used the 4M files for both the bios and the nvram. Thanks!

**Comment by donvipre** [comment]
Hey guys,
I stumbled over this also in September and for Fedora this is what worked for me:
1. in your VM XML you will find the <loader> and <nvram> sections
1. create backup of everything, but at least `/var/lib/libvirt/qemu/nvram/win11_VARS.fd` filename depends on the VM you want to alter.
    This includes BitLocker Recovery Key.
2. `cp /usr/share/edk2/ovmf-4m/OVMF_VARS.fd /var/lib/libvirt/qemu/nvram/win11_VARS.fd` again filename depends on the VM file.
3. alter the XML to below values:
    ```
    <loader readonly="yes" secure="yes" type="pflash">/usr/share/edk2/ovmf-4m/OVMF_CODE.secboot.fd</loader>
    <nvram>/var/lib/libvirt/qemu/nvram/win11_VARS.fd</nvram>```
4. VM should now boot properly and the Windows Update should disappear.

Thanks to @BentHaase for finding out!

**Reply by svarshavchik** [reply]
Huh, it does seem, indeed, that edk2-ovmf-20220826gitba0e0e4c6a17-1.fc36.noarch includes the 4m images, now.

Based on earlier messages here it should be sufficient to only reconfigure the XML to point to the 4m images, no need to reset the vars file.

**Reply by kjkent** [reply]
@donvipre you are a hero

**Reply by svarshavchik** [reply]
I can report that the 4m images also worked for me; however I could not keep my existing VARS.fd file, and had to reset mine's to the 4m one. I don't use bitlocker, so I can't comment on that. The only effect of the VARS.fd file reset is that my boot resolution went back to the default one, and I had to reset it back to what it was. That was the only effect on me.

**Reply by kjkent** [reply]
@svarshavchik ah yes, I should have mentioned this was also the case for me.

**Reply by Windows-Guy** [reply]
> Hey guys, I stumbled over this also in September and for Fedora this is what worked for me:
> 
> 1. in your VM XML you will find the  and  sections
> 2. create backup of everything, but at least `/var/lib/libvirt/qemu/nvram/win11_VARS.fd` filename depends on the VM you want to alter.
>    This includes BitLocker Recovery Key.
> 3. `cp /usr/share/edk2/ovmf-4m/OVMF_VARS.fd /var/lib/libvirt/qemu/nvram/win11_VARS.fd` again filename depends on the VM file.
> 4. alter the XML to below values:
>    ```
>    <loader readonly="yes" secure="yes" type="pflash">/usr/share/edk2/ovmf-4m/OVMF_CODE.secboot.fd</loader>
>    <nvram>/var/lib/libvirt/qemu/nvram/win11_VARS.fd</nvram>```
>    ```
> 5. VM should now boot properly and the Windows Update should disappear.
> 
> Thanks to @BentHaase for finding out!

Im on fedora with win11 too. I tried this and it worked. But i checked and my vm has secureboot off now. Thanks don but ya might wanna check that because you wont get feature updates now.

**Reply by donvipre** [reply]
@Windows-Guy: sorry for the late response, but where did you see the "no secure boot"?
I didn't notice it and would assume that the VM runs well.

**Reply by Windows-Guy** [reply]
I think its in windows settings security devices in win 11. Just google hiw do i tell if secure boot is on in windows whatever. But the gist of it is that fedoras not doing something right yet w those 4m uefis. And when you do what you and i did, that kb5012170 goes away nit because it installed but because secure boot is now off. So the problem didnt go away. Upcoming updates like that one wont be a problem because secure boot is off. What will be a problem is when a feature update like 22h2 comes out. Windows will say your machine doesnt qualify for it. My work arround until this works by default on fedora was to go back to mint. 21.1 came out after i tried this on fedora37 and as long as i picked the uefi that said blah blah 4M.ms.  Ir something like that, it worked. I had secboot on and it installed the 2170 update like it was a real machine. Now i should get updates like that and qualify for yearly big updates. Hope this helps

**Reply by Windows-Guy** [reply]
Just thought you should check yours to see if secure boot is enabled. If it is youre done. But if it did what mine did, the problem just got pushed a year out when itll be a bigger problem to start over

**Reply by CrazyLinuxNerd** [reply]
> I think its in windows settings security devices in win 11. Just google hiw do i tell if secure boot is on in windows whatever. But the gist of it is that fedoras not doing something right yet w those 4m uefis. And when you do what you and i did, that kb5012170 goes away nit because it installed but because secure boot is now off. So the problem didnt go away. Upcoming updates like that one wont be a problem because secure boot is off. What will be a problem is when a feature update like 22h2 comes out. Windows will say your machine doesnt qualify for it. My work arround until this works by default on fedora was to go back to mint. 21.1 came out after i tried this on fedora37 and as long as i picked the uefi that said blah blah 4M.ms. Ir something like that, it worked. I had secboot on and it installed the 2170 update like it was a real machine. Now i should get updates like that and qualify for yearly big updates. Hope this helps

This is happening because the wrong VARS file is being copied. You don't want the 4M version of "OVMF_VARS.fd" for secure boot, you want "OVMF_VARS_4M.ms.fd". I use the ones from the deb file however, so I'm not sure on the exact location from the original guide, but in theory it is probably here:

cp /usr/share/edk2/ovmf-4m/OVMF_VARS.ms.fd

Don't quote me on that though, it's just a guess!

In any case, using the "ms" fd file should re-activate your secure boot (if nothing else was/is broken).

**Comment by mswal2846** [comment]
I did as suggested by @donvipre but when I attempt to boot up the virtual machine I only get a blank screen.  The CPU peaks so something is happening but nothing else can be seen.  I verified the ownership and permissions were the same but that didn't make any difference.

So then I thought I would reinstall Win11 and pick the /usr/share/edk2/ovmf-4m lib for the Hypervisor setup but when I attempt to begin the install I get an error message about PCI versus PCIe. 

I'm very much a noob so my apologies if I'm asking/doing stupid questions

**Comment by brianwoo** [comment]
I had the same problem and my host machine is running Mint 21 (Ubuntu 22.04 LTS).  I can confirm that changing from OVMF_CODE.fd to OVMF_CODE_4M.fd in the XML solved the issue.

**Reply by sladiri** [reply]
> I had the same problem and my host machine is running Mint 21 (Ubuntu 22.04 LTS). I can confirm that changing from OVMF_CODE.fd to OVMF_CODE_4M.fd in the XML solved the issue.

Similar here, Fedora 37, have a VM without secure-boot, or Bitlocker. There was no OVMF file set in XML.

It did not let me apply with the path setting added due to an auto-select firmware setting. I could not find how to turn that off, so I recreated the VM, reused the disk. I also had to use a backup of the VRAM file, because it got deleted with the old VM afterwards.

Booted right away without issues.

(In Fedora 37 the 4M images are in a directory next to the small images.)

**Comment by Windows-Guy** [comment]
I want 3 things in my win11 kvm
Secureboot
Tpm2
And the ability to do these dbx updates like kb5012170

so ive been debating between mint and fedora
I could not get these 3 in mint 21
So i started over in fedora 37
And ive tried every combo of uefi and tpm
And what don did
And i can get everything but kb5012170 to update (or i get it and realize secboot is off)

id just did a fresh install of mint 21.1
Installed virt manager from software
Rebooted
Started a new kvm
Picked crb tpm 2
And ovmf????-4m.ms.fd
And win11 came up with tpm2, secboot on,
And kb5012170 dbx updated

im done
Id love to be on fedora
I think its more secure
But i keep going back to mint
Because i dont have to fight it
To make it work the way i want

**Reply by kjkent** [reply]
@Windows-Guy this reads like a poem -- I love it.

If you really want Fedora, I've used KVM/QEMU on it as a host, though, with Win11 & Win10 (when I realised how bloaty Win11 is) as guests. My method was to use  `ovmf-4m-secboot.fd` (can't remember the exact name) in situ, but then copy the corresponding VARS file to a writable location and include that in the config.

I'm now using openSUSE and really enjoying it. I moved over my Win10 VM but forgot to copy the config so redid it from scratch, just importing the `.qcow2` file and it worked immediately. Here's my XML for that:

```xml
  <os>
    <type arch="x86_64" machine="pc-q35-7.1">hvm</type>
    <loader readonly="yes" type="pflash">/usr/share/qemu/ovmf-x86_64-ms-4m-code.bin</loader>
    <nvram>/home/kjkent/vm/win10ltsc2021.vars.secboot.fd</nvram>
    <boot dev="hd"/>
  </os>
```

**Reply by Windows-Guy** [reply]
I just tried fedora 38, it still doesnt work by some default pick. It seems that the 4m edk2 stuff id there but you cant pick it in virt manager. In any case, it works perfectly in mint 21.1. Im officially done with this project. I think the problem with bleeding edge distros is it take so much time to integrate all the new stuff that the project as a whole never gets any quality control because its a moving target every 6 months.

**Reply by svarshavchik** [reply]
This needs to be resolved in the virt-manager package. I have created a bug against virt-manager, to add the 4m images, and, yes, nothing have happened on it, yet.

**Comment by mswal2846** [comment]
I was finally able to get this to work under Fedora 37 using Virtual Machine.  I followed this set of directions to reinstall Windows 11 in a Virtual Machine:  https://getlabsdone.com/how-to-install-windows-11-on-kvm/#tpm-and-secureboot

When I got to step 8. Enable Secure-Boot, I had to do the follow:
1. Go to /usr/share/edk2/ - I had two subdirectories:  ovmf & ovmf-4m
2. Rename ovmf to "ovmf-old" (or whatever you want to rename it to)
3. Rename ovmf-4m to ovmf
4. Proceed with step 8 in the above manual but be sure to select "/usr/share/OVMF/OVMF_CODE.4m.secboot.fd" in the Hypervisor details (second option in step 8).
5. Then proceed through the rest of the steps.

When Windows 11 was up and running, all updates have applied successfully.

**Reply by svarshavchik** [reply]
Let's not do that, let's not rename directories like that. Otherwise, next time you run `dnf update` and an ed2k-obmf package gets updated you are guaranteed that much hillarity will ensue.

It's much safer to `virsh edit` the domain and make that change directly to the XML. For reference, the relevant portion of my domain looks like this:

```
  <os>
    <type arch='x86_64' machine='pc-q35-5.2'>hvm</type>
    <loader readonly='yes' secure='yes' type='pflash'>/usr/share/edk2/ovmf-4m/OVMF_CODE.secboot.fd</loader>
    <nvram>/var/lib/libvirt/qemu/nvram/WINDOWS10-EFI_VARS.fd</nvram>
    <boot dev='hd'/>
    <bootmenu enable='yes'/>
  </os>
```
If you're setting up a new domain, go ahead and set it up in libvirt, but don't start it, and then use `virsh edit` to fix it.

The right solution is to get virt-manager to offer the option to configure ovmf-4m. I have a bug collecting dust, somewhere, in Bugzilla, about that...

**Reply by svarshavchik** [reply]
Forgot to mention that after setting up a domain, you'll still need to manually copy the OVMF-VARS to overwrite the one that libvirt installs.

**Reply by mswal2846** [reply]
Ahh, right ... I did say previously that I was new to this.  Thanks for the correction!

**Reply by sladiri** [reply]
I agree and these are the steps I did while setting up a new domain/VM:
(I could not change an existing domain, because it complained that a path could not be set because of a "auto-select firmware setting". Could not find out how to disabled "auto-select firmware")

- ticked the "customize before install" option:
- I specified a OVMF_CODE file in settings instead of the default.
- Then I had to start it once and shut it down.
- Then I edited the XML and changed the path to the file to use `ovmf-4m` instead of `ovmf`.

**Reply by CLNRMN** [reply]
Thanks for directing me in the right direction. But with the `ovmf-4m` files in the package `edk2-ovmf-20230301gitf80f052277c8-1.fc37.noarch` i was not able to boot win11. It was stuck in the `TianoCore` boot loader screen. 

I've tested it then with the latest `OVMF_CODE.secboot.fd` from the debian package and it was working under fedora 37

**Reply by bzeiss** [reply]
Using the .fd from the Debian package worked for me too.

**Comment by DutchTurtle** [comment]
Im pretty new to virtualization in fedora and i was not able to start the windows installer for a new VM with qemu. even after reading this thread. however when i downgraded edk2-ovmf (sudo dnf downgrade edk2-ovmf)  i was able to run a new VM + installer withouth a problem. Hope that helps someone!

**Reply by ghost** [reply]
That fixed it for me - thanks!

**Comment by mt190502** [comment]
Same issue on my machine

Hello. I am using Fedora 38. I cannot run current windows 10 and windows 11 images in UEFI mode. 
What I've Tried:
- https://github.com/tianocore/edk2/discussions/3221#discussioncomment-4470342
- https://github.com/tianocore/edk2/discussions/3221#discussioncomment-5195657
- https://github.com/tianocore/edk2/discussions/3221#discussioncomment-5501246

**Reply by mt190502** [reply]
![image](https://user-images.githubusercontent.com/62564400/234611309-85cbd0eb-4715-4ddb-bc93-1edb6a5e18d9.png)

**Reply by mswal2846** [reply]
When you  say "run" do you mean that Windows 10 and 11 were installed and now that you upgraded to Fedora 38 you are unable to run them?  Or are you saying that you are unable to install Windows 10 or 11 under Fedora 38?

I installed Windows 11 under Fedora 37 and even after the upgrade I am able to start Windows 11.

So I am going to assume that you are unable to install Windows 11.  I followed this guide:  

https://getlabsdone.com/how-to-install-windows-11-on-kvm/#tpm-and-secureboot

There are a few issues with these directions due to QEMU-KVM installer.  But this guide will get you started.  Step 8 (Enable Secure-Boot) you will want to use the section with "If you are on latest version of secureboot, you will see UEFI x86_64:/usr/share/OVMF/OVMF_CODE.4m.secboot.fd ."

Come back with any issues when you start up and I will try to help.

**Reply by mt190502** [reply]
> When you say "run" do you mean that Windows 10 and 11 were installed and now that you upgraded to Fedora 38 you are unable to run them? Or are you saying that you are unable to install Windows 10 or 11 under Fedora 38?
> 
> I installed Windows 11 under Fedora 37 and even after the upgrade I am able to start Windows 11.
> 
> So I am going to assume that you are unable to install Windows 11. I followed this guide:
> 
> https://getlabsdone.com/how-to-install-windows-11-on-kvm/#tpm-and-secureboot
> 
> There are a few issues with these directions due to QEMU-KVM installer. But this guide will get you started. Step 8 (Enable Secure-Boot) you will want to use the section with "If you are on latest version of secureboot, you will see UEFI x86_64:/usr/share/OVMF/OVMF_CODE.4m.secboot.fd ."
> 
> Come back with any issues when you start up and I will try to help.

I can't install windows 10/11 with virt-manager in Fedora 38.

 In the link you sent, i440fx is selected, but my system is not compatible with i440fx. I am getting """The PCI controller with index='0' must be model='pci-root' for this machine type, but model='pcie-root' was found instead""" error. When I asked this a few places, I saw comments that the Q35 is more comprehensive and better. 

After selecting Q35, the files in the /usr/share/edk2/ovmf folder appear in the BIOS. I first select the /usr/share/edk2/ovmf/OVMF_CODE.fd file. Then in the xml file I change the path to /usr/share/edk2/ovmf-4m/OVMF_CODE.fd

![image](https://user-images.githubusercontent.com/62564400/234632560-97579a83-f71f-49c8-86de-23e59ccdf8f8.png)

**Reply by mt190502** [reply]
XML File: https://paste.centos.org/view/raw/cd029b57

**Reply by mswal2846** [reply]
This error ""The PCI controller with index='0' must be model='pci-root' for this machine type, but model='pcie-root' was found instead"" is due to a bug in Virtual-manager (which I was alluding to in my initial response).  I encountered this too and found this info (https://github.com/virt-manager/virt-manager/issues/398):

_You  select win10, so virt-manager defaults to q35. When you enter the 'Customize' wizard, virt-manager pre-populate the XML with a bunch of pcie controller bits, to enable PCIe hotplug on q35. You are trying to switch back to machine=pc, but then libvirt says 'hey all those pcie bits in the XML aren't valid for machine=pc'. Yes that's a bug in virt-manager

You can work around this by deleting all the <controller type='pci* bits in the XML first, with the XML editor tab in the virt-manager Overview section, and then change q35 to pc. But yeah virt-manager needs to be fixed too_

You'll want to do that above prior to booting up

**Reply by mt190502** [reply]
> You can work around this by deleting all the <controller type='pci* bits in the XML first, with the XML editor tab in the virt-manager Overview section, and then change q35 to pc. But yeah virt-manager needs to be fixed too_
> 
> You'll want to do that above prior to booting up

Same issue :(

![image](https://user-images.githubusercontent.com/62564400/234643248-fb991578-1eab-4e59-9b16-6996ef6112a9.png)

![image](https://user-images.githubusercontent.com/62564400/234641744-0e565e0a-3542-4162-aaaa-c4a23f640399.png)

**Reply by mswal2846** [reply]
So...
I attempted to setup a new Windows 11 virtual machine just to see what's going on and I am having no success either.  I can get to the begin installation step, the console asks to go into CD or DVD selection but then seems to spin.  I'll play around some more to see if I can figure it out.

**Reply by mswal2846** [reply]
I think until Virtual-Machine Manager fixes the secure boot process this is broke under Fedora 38.  You can try what others have documented above (e.g., DutchTurtle's post from 3 weeks ago).  I can't do that without running the risk of breaking my existing windows 11 virtual machine.  svarshavchik has opened a bug on this issue as documented above.

Sorry for not being able to provide much help.

**Reply by mt190502** [reply]
> I think until Virtual-Machine Manager fixes the secure boot process this is broke under Fedora 38. You can try what others have documented above (e.g., DutchTurtle's post from 3 weeks ago). I can't do that without running the risk of breaking my existing windows 11 virtual machine. svarshavchik has opened a bug on this issue as documented above.
> 
> Sorry for not being able to provide much help.

np, thanks for helps
